src/FilterFormatManager.php

```diff
@@ -0,0 +1,166 @@
+   * @return array
+   *   The list of filter id's for a given filter format or an empty list.
+   */
+  public function getFilterIds(FilterFormat $filter_format): array {
+    $ids = $filter_format->filters()->getInstanceIds();
+    return is_array($ids) && !empty($ids) ? array_keys($ids) : [];
+  }
+
+  /**
+   * Verifies whether a filter id is valid for a given filter format or not.
+   *
+   * @param string $filter_id
+   *   The filter id.
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format.
+   *
+   * @return bool
+   *   The answer to whether a filter id is valid for the given filter format or
+   *   not.
+   */
+  public function isValidFilterId(string $filter_id, FilterFormat $filter_format): bool {
+    return in_array($filter_id, $this->getFilterIds($filter_format));
+  }
+
+  /**
+   * Returns a list of enabled filters for a given filter format.
+   *
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format object.
+   *
+   * @return \Drupal\filter\FilterPluginCollection|null
+   *   The list of enabled filters for the given filter format.
+   */
+  public function getEnabledFilters(FilterFormat $filter_format): ?FilterPluginCollection {
+    /** @var \Drupal\filter\FilterPluginCollection|null $filters */
+    $filters = $filter_format->filters();
+    if (!($filters instanceof FilterPluginCollection) || !($filters->count() > 0)) {
+      return NULL;
+    }
+    foreach ($filters as $filter) {
+      $configuration = $filter->getConfiguration();
+      $status = is_array($configuration) && array_key_exists('status', $configuration) ? $configuration['status'] : FALSE;
+      if ($status !== TRUE) {
+        $filters->removeInstanceId($filter->getPluginId());
+      }
+    }
+
+    return $filters;
+  }
+
+  /**
+   * Returns a list of allowed HTML tags for a given filter format.
+   *
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format object.
+   *
+   * @return array
+   *   The list of allowed HTML tags for the given filter format if any, else an
+   *   empty array.
+   */
+  public function getAllowedTags(FilterFormat $filter_format): array {
+    /** @var array $html_restrictions */
+    $html_restrictions = $filter_format->getHtmlRestrictions();
+    /** @var string[] $allowed_tags */
+    $allowed_tags = is_array($html_restrictions) && array_key_exists('allowed', $html_restrictions) ? array_keys($html_restrictions['allowed']) : [];
+
+    return $allowed_tags;
+  }
+
+  /**
+   * Verifies if a tag is among the allowed HTML tags for a given filter format.
+   *
+   * The verification is case-insensitive.
+   *
+   * @param string $tag
+   *   The tag to verify.
+   * @param \Drupal\filter\Entity\FilterFormat $filter_format
+   *   The filter format object.
+   *
+   * @return bool
+   *   The result of the verification operation.
+   */
+  public function isAllowedTag(string $tag, FilterFormat $filter_format): bool {
```

Comment:
isTagAllowed()

Reply:
**Status:** Fixed
The method has been renamed to `isTagAllowed()`.